### ISSUE TITLE
obol: issues with special characters/missing fields

### ISSUE THREAD

**Issue by themisa** [clustervision/trinity#419]
```
[1:13:58 PM] Máni Maríus Xander: hmm, something also went wrong when using the obol command. Maybe it was the missing given name or the accented character. Fixing both of those and creating a new user worked. Here is the dump
[1:14:11 PM] Máni Maríus Xander: [root@login ~]# obol user list
[root@login ~]# obol user add --password fiskur --cn Máni --sn Xander --groups power-users mani
('expected a string in the list', None)
Traceback (most recent call last):
  File "/usr/sbin/obol", line 305, in <module>
    fun(**args_d)
  File "/usr/sbin/obol", line 42, in user_add
    conn.add_s(dn, add_record)
  File "/usr/lib64/python2.7/site-packages/ldap/ldapobject.py", line 194, in add_s
    msgid = self.add(dn,modlist)
  File "/usr/lib64/python2.7/site-packages/ldap/ldapobject.py", line 191, in add
    return self.add_ext(dn,modlist,None,None)
  File "/usr/lib64/python2.7/site-packages/ldap/ldapobject.py", line 176, in add_ext
    return self._ldap_call(self._l.add_ext,dn,modlist,RequestControlTuples(serverctrls),RequestControlTuples(clientctrls))
  File "/usr/lib64/python2.7/site-packages/ldap/ldapobject.py", line 99, in _ldap_call
    result = func(*args,**kwargs)
TypeError: ('expected a string in the list', None)
[root@login ~]# obol user add --password fiskur --cn Mani --sn Xander --groups power-users mani
{'desc': 'Already exists'}
Traceback (most recent call last):
  File "/usr/sbin/obol", line 305, in <module>
    fun(**args_d)
  File "/usr/sbin/obol", line 26, in user_add
    conn.add_s(dn, add_record)
  File "/usr/lib64/python2.7/site-packages/ldap/ldapobject.py", line 195, in add_s
    return self.result(msgid,all=1,timeout=self.timeout)
  File "/usr/lib64/python2.7/site-packages/ldap/ldapobject.py", line 458, in result
    resp_type, resp_data, resp_msgid = self.result2(msgid,all,timeout)
  File "/usr/lib64/python2.7/site-packages/ldap/ldapobject.py", line 462, in result2
    resp_type, resp_data, resp_msgid, resp_ctrls = self.result3(msgid,all,timeout)
  File "/usr/lib64/python2.7/site-packages/ldap/ldapobject.py", line 469, in result3
    resp_ctrl_classes=resp_ctrl_classes
  File "/usr/lib64/python2.7/site-packages/ldap/ldapobject.py", line 476, in result4
    ldap_result = self._ldap_call(self._l.result4,msgid,all,timeout,add_ctrls,add_intermediates,add_extop)
  File "/usr/lib64/python2.7/site-packages/ldap/ldapobject.py", line 99, in _ldap_call
    result = func(*args,**kwargs)
ldap.ALREADY_EXISTS: {'desc': 'Already exists'}
[root@login ~]# obol user list
[root@login ~]# obol user add --password fiskur --cn Mani --givenName Mani --sn Xander --groups power-users mani2
[root@login ~]# obol user list
mani2
```

**Comment by hansthen** [comment]
This problem was not with the missing characters, but with the missing cn. Because of this the group was created, but not the user.

**Comment by hansthen** [comment]
Fixed in 9cac053aa00e2c3805972be3c1587fffd0f49afe

**Comment by themisa** [comment]
**Version:** r6_fixes b8c25e1
**Environment:** tmaster

Users can now be created without mandatory fields
```
[root@login home]# obol -w system user add --password 123 john
[root@login home]# obol -w system user list
jane
john
[root@login home]# ls /home
jane  munge  slurm
```
